tests/unit/test_spack.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Test minimal snap."""
+
+import logging
+import os
+import pathlib
+import re
+import shlex
```

Comment:
I don't think you need to use `shlex` here because the commands you are executing with subprocess are relatively simple. If they were more complex or you were executing commands based on user input then I would support using shlex, but I don't think you need it in this case.

Reply:
Fair, I just like the readability while I'm hacking away.